java/sage/Carny.java

```diff
@@ -953,188 +1154,445 @@ private void stdProcessing()
+                ((ArrayList) destAirings).ensureCapacity(airSetExpand);
+
+              destAirings = firstCallback.watchedPotsToClear;
+              if (destAirings instanceof ArrayList)
+                ((ArrayList) destAirings).ensureCapacity(watchedPotsToClearExpand);
 
-        Float wp = newWPMap.get(agePots[j]);
-        boolean replaceInMap = false;
-        if (wp == null || (currAgent.watchProb > wp.floatValue()))
-          replaceInMap = true;
-        if (wp != null && currAgent.watchProb == wp.floatValue() && currAgent.isFavorite())
+              destAirings = firstCallback.newMustSeeSet;
+              if (destAirings instanceof ArrayList)
+                ((ArrayList) destAirings).ensureCapacity(newMustSeeSetExpand);
+
+              List<WPCauseValue> destCauses = firstCallback.newWPCauses;
+              if (destCauses instanceof ArrayList)
+                ((ArrayList) destCauses).ensureCapacity(newWPCausesExpand);
+
+              for (int j = 0; j < agentWorkerFuturesSize; j++)
+              {
+                callback = callbacks[j];
+                if (callback == firstCallback)
+                  continue;
+
+                // Since we allowed the workers to build their own idea of watch potential and
+                // causes, we need to consolidate those maps here. This is still faster because
+                // the threads are not fighting each other over the same map and it removes a
+                // synchronization requirement.
+                List<WPCauseValue> srcCauses = callback.newWPCauses;
+                for (int k = 0, size = srcCauses.size(); k < size; k++)
+                {
+                  WPCauseValue currCauseValue = srcCauses.get(k);
+                  WPCauseValue mappedCauseValue = firstCallback.addOrReturnWPCauseValue(currCauseValue);
+                  Airing agentPot = currCauseValue.airing;
+
+                  // We just added this agent.
+                  if (mappedCauseValue == null)
+                  {
+                    if (agentPot.isWatchedForSchedulingPurpose())
+                      firstCallback.addWatchedPotsToClear(agentPot);
+                  }
+                  else
+                  {
+                      // This agent doesn't override the current agent.
+                    if (!mappedCauseValue.compareAndReplace(currCauseValue.agent))
+                    {
+                      // They are the same wp, so they might be out of order. We are mostly
+                      // interested in favorites which is why we don't do this for 0 WP, but for
+                      // higher values than 0, we try to keep things consistent.
+                      if (mappedCauseValue.wp != MIN_WP && mappedCauseValue.wp == currCauseValue.wp)
+                      {
+                        // Fix the order of this entry so we have the same results every time.
+                        if (currCauseValue.agent.createTime > mappedCauseValue.agent.createTime)
```

Comment:
So does AGENT_SORTER give you those bad results then? I'm fine with you leaving it as creation time if that matches the closest...but I'm a little unsure of the results from my suggestion based on the multiple comments. :)